### PR TITLE
Radios now make noises when you use them

### PR DESCRIPTION
**Pull request by VictorVow**
## About The Pull Request
Upstreamed from NSV. Adds a sound effect for when you use a radio. 

## Why It's Good For The Game
Aids with the ambience and audio feedback feels nice when using a radio. 

## Changelog
:cl:
add: Radios now make noises when you use them
/:cl:

Ports:
https://github.com/BeeStation/NSV13/pull/8

### CODE REVIEW

**Comment by Moccha-Bee** [comment]
> ## About The Pull Request
> Upstreamed from NSV. Adds a sound effect for when you use a radio.
> 
> ## Why It's Good For The Game
> Aids with the ambience and audio feedback feels nice when using a radio.
> 
> ## Changelog
> 🆑
> add: Radios now make noises when you use them
> /🆑
> 
> Ports:
> [BeeStation/NSV13#8](https://github.com/BeeStation/NSV13/pull/8)

give us sound or bad pr insta close 😳

**Comment by VictorVow** [comment]
It's in the files changed lazy ass

[radio1.ogg](https://raw.githubusercontent.com/BeeStation/BeeStation-Hornet/f65e60ea92967ea3ccf68fac62e2421672737167/sephora/sound/effects/radio1.ogg)
[radio2.ogg](https://raw.githubusercontent.com/BeeStation/BeeStation-Hornet/f65e60ea92967ea3ccf68fac62e2421672737167/sephora/sound/effects/radio2.ogg)

**Comment by Moccha-Bee** [comment]
> It's in the files changed lazy ass
> 
> [radio1.ogg](https://raw.githubusercontent.com/BeeStation/BeeStation-Hornet/f65e60ea92967ea3ccf68fac62e2421672737167/sephora/sound/effects/radio1.ogg)
> [radio2.ogg](https://raw.githubusercontent.com/BeeStation/BeeStation-Hornet/f65e60ea92967ea3ccf68fac62e2421672737167/sephora/sound/effects/radio2.ogg)

i can't open the ogg files 😔

**Comment by Moccha-Bee** [comment]
> > It's in the files changed lazy ass
> > [radio1.ogg](https://raw.githubusercontent.com/BeeStation/BeeStation-Hornet/f65e60ea92967ea3ccf68fac62e2421672737167/sephora/sound/effects/radio1.ogg)
> > [radio2.ogg](https://raw.githubusercontent.com/BeeStation/BeeStation-Hornet/f65e60ea92967ea3ccf68fac62e2421672737167/sephora/sound/effects/radio2.ogg)
> 
> i can't open the ogg files 😔

okay it sounds nice i guess.

**Comment by ghost** [comment]
This was actually tried once already and people didn't really like it, (#121) i suppose a second try couldn't hurt though.

**Comment by Moccha-Bee** [comment]
> This was actually tried once already and people didn't really like it, (#121) i suppose a second try couldn't hurt though.

why did you leave the discord bro, i miss you...

**Comment by Trash0077** [comment]
Audio feedback is always nice, good pr

**Comment by utilbird** [comment]
Kmc is a port magnet

File: code/game/objects/items/devices/radio/radio.dm [hunk 1]
```diff
@@ -210,6 +211,9 @@
 		return
 	if(!M.IsVocal())
 		return
+	if(radio_sounds.len) //Sephora - Radios make small static sounds now.
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/radio/radio.dm]: Use LAZYLEN

File: code/game/objects/items/devices/radio/radio.dm [hunk 2]
```diff
@@ -210,6 +211,9 @@
...
+		var/sound/radio_sound = pick(radio_sounds)
+		playsound(M.loc, radio_sound, 50, 1)
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/radio/radio.dm:217]: I think this should be played to the person sending the message but not the people around them

File: code/game/objects/items/devices/radio/radio.dm [hunk 3]
```diff
@@ -39,6 +39,7 @@
 	var/syndie = FALSE  // If true, hears all well-known channels automatically, and can say/hear on the Syndicate channel.
 	var/list/channels = list()  // Map from name (see communications.dm) to on/off. First entry is current department (:h).
 	var/list/secure_radio_connections
+	var/list/radio_sounds = list('sephora/sound/effects/radio1.ogg','sephora/sound/effects/radio2.ogg') //Sephora - Radios make small static noises now
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/radio/radio.dm]: The paths should just be sound/... rather than sephora/sound...

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/radio/radio.dm (hunk 2 above)
Comment (DatBoiTim) [reply code/game/objects/items/devices/radio/radio.dm:217]: Internal syndicate radio go _brrr_

**Comment by GoatMonarch** [comment]
Fair warning, but implant radios also makes a noise when used. Making it quite obvious you're using one.
Unless this has changed, you're porting the same issue from downstream.

**Comment by DatBoiTim** [comment]
> Fair warning, but implant radios also makes a noise when used. Making it quite obvious you're using one.
> 
> Unless this has changed, you're porting the same issue from downstream.

PowerfulBacon(Repo Maintainer) already brought that up to him.

But yes, internal syndicate radio go _brrrrrr_

File: code/game/objects/items/devices/radio/radio.dm (hunk 2 above)
Comment (francinum) [reply code/game/objects/items/devices/radio/radio.dm:217]: Internal radios should have the sound list nulled.

File: code/game/objects/items/devices/radio/radio.dm (hunk 1 above)
Comment (VictorVow) [reply code/game/objects/items/devices/radio/radio.dm]: Done

File: code/game/objects/items/devices/radio/radio.dm (hunk 2 above)
Comment (VictorVow) [reply code/game/objects/items/devices/radio/radio.dm:217]: Internal radios are now silent

File: code/game/objects/items/devices/radio/radio.dm (hunk 3 above)
Comment (VictorVow) [reply code/game/objects/items/devices/radio/radio.dm]: Fixed

**Comment by VictorVow** [comment]
Should be ready.

File: code/game/objects/items/devices/radio/radio.dm [hunk 4]
```diff
@@ -39,6 +39,8 @@
 	var/syndie = FALSE  // If true, hears all well-known channels automatically, and can say/hear on the Syndicate channel.
 	var/list/channels = list()  // Map from name (see communications.dm) to on/off. First entry is current department (:h).
 	var/list/secure_radio_connections
+	var/radio_silent = FALSE // If true, radio doesn't make sound effects (ie for Syndicate internal radio implants)
+	var/list/radio_sounds = list('sound/effects/radio1.ogg','sound/effects/radio2.ogg') //Sephora - Radios make small static noises now
```
Comment (ike709) [inline code/game/objects/items/devices/radio/radio.dm]: Remove the list and just `pick()` these two where they're used.

Readability < memory usage. Plus you shouldn't init a list at the type declaration level anyways.

Comment (VictorVow) [reply code/game/objects/items/devices/radio/radio.dm]: Fixed

**Comment by VictorVow** [comment]
Should be ready now.

**Comment by ZombiesVsPlants** [comment]
this might have been alright for the average player but holy FUCK is it annoying while playing ai since you're constantly talking and organizing over the radio. Please give us an option/preference to turn it off.

**Pull request merged**